opentreemap/treemap/audit.py

```diff
@@ -812,17 +817,20 @@ def _get_writeable_perms_set(self, user, direct_only=False):
             perm_set = {perm.field_name for perm in perms
                         if perm.allows_writes}
 
-        return perm_set.union(self._get_joint_writeable_fields(user))
+        return perm_set.union(self._get_joint_writable_fields(user))
 
-    def _get_joint_writeable_fields(self, user):
-        # If any field on any model is writable in any capacity, read
+    def _get_joint_writable_fields(self, user):
+        # If any field on *any* model is writable in any capacity, read
         # a class property to get the set of field names that are also
-        # writable.
+        # writable. The idea is that if a tree is updated, the plot's
+        # `updated_at` must get updated, even if the role has no write
+        # access on the plot.
+        field_perms = field_permissions(user, self.get_instance())
         can_write_anything = bool({perm.field_name for perm
-                                   in permissions(user, self.get_instance())
+                                   in field_perms
                                    if perm.allows_writes})
         if can_write_anything:
-            return getattr(type(self), 'joint_writable', set())
+            return self._joint_writable
         else:
             return set()
```

Comment:
I think we should remove the `can_write_anything` check (and update the comment). It complicates the logic and doesn't add significant safety. It raises the possibility that I could have create permission for e.g. plots but no permission to modify fields, and thus be prevented from creating a plot because `id` was not in the set of writable fields. `id`, `updated_at`, `geom`, and `tree.plot` should truly be exempt from field permissions.